lib/matplotlib/patches.py

```diff
@@ -71,7 +79,7 @@ def __init__(self, *,
         if joinstyle is None:
             joinstyle = JoinStyle.miter
 
-        self._hatch_color = colors.to_rgba(mpl.rcParams['hatch.color'])
+        self.set_hatch_color = False
```

Comment:
Hi, I was trying to resolve the original issue - which was distinguishing between user-set-black and default black.
I'll try to explain a little bit about how I am doing this. 
So, basically the behavior that I intend to achieve is that hatchcolor should default to edgecolor and then rcParams. I am trying to do this on the patch object so that this is automatically applied to all functions which return a patch object. 
These values are automatically being set using `_internal_update(kwargs)`. 

So, the issue that I was facing was that if edgecolor was set after hatchcolor, then `self._hatch_color` was being overridden by edgecolor because of the `_internal_update(kwargs)` method because we were setting hatchcolor from `_set_edgecolor`.

So, using this variable it is helping me check whether the hatchcolor is set by user or not. If it isn't set by user, only then the edgecolor will be able to set the hatchcolor. 

Also, the default value of hatchcolor will still be black but `set_hatch_color` will still be false so we know that the user has not set the hatchcolor. So, the edgecolor function can now update the `self._hatch_color`.


```py
import matplotlib.pyplot as plt
import numpy as np

x = np.random.normal(170, 10, 250)

plt.hist(x, hatch="/", color="yellow",  hatchcolor="blue", edgecolor="red")
plt.show() 
```
The problem in this code was that because of `_internal_update(kwargs)`, the hatchcolor was being set first and then the edgecolor was overriding the hatchcolor. So, I introduced another variable set_hatch_color.

I don't know if this is clear enough. Let me know if you have any doubts.